apps/emqx_durable_storage/src/emqx_ds_replication_layer.erl

```diff
@@ -301,14 +322,35 @@ do_next_v1(DB, Shard, Iter, BatchSize) ->
 -spec do_add_generation_v2(emqx_ds:db()) -> ok | {error, _}.
 do_add_generation_v2(DB) ->
     MyShards = emqx_ds_replication_layer_meta:my_owned_shards(DB),
-
     lists:foreach(
         fun(ShardId) ->
             emqx_ds_storage_layer:add_generation({DB, ShardId})
         end,
         MyShards
     ).
 
+-spec do_list_generations_with_lifetimes_v3(emqx_ds:db()) -> ok | {error, _}.
```

Comment:
Q: Can we pass the shard as an argument and do aggregation on the caller side? It feels like a more fine-grained API.